crypto/kms/pubkeystore.go

```diff
@@ -135,54 +144,163 @@ func GetPublicKey(id proto.NodeID) (publicKey *ec.PublicKey, err error) {
 			return ErrKeyNotFound
 		}
 
-		publicKey, err = ec.ParsePubKey(byteVal, ec.S256())
+		// error log will be done at the wrapper func
+		nodeInfoTypes := &types.Node{}
+		err = pb.Unmarshal(byteVal, nodeInfoTypes)
+		if err == nil {
+			publicKey, err := ec.ParsePubKey(nodeInfoTypes.PublicKey.PublicKey, ec.S256())
+			if err == nil {
+				nodeInfo = &proto.Node{
+					ID:        proto.NodeID(nodeInfoTypes.ID.NodeID),
+					Addr:      nodeInfoTypes.Addr,
+					PublicKey: publicKey,
+					Nonce: mine.Uint256{
+						A: nodeInfoTypes.Nonce.A,
+						B: nodeInfoTypes.Nonce.B,
+						C: nodeInfoTypes.Nonce.C,
+						D: nodeInfoTypes.Nonce.D,
+					},
+				}
+			}
+		}
+
 		return err // return from View func
 	})
+	if err != nil {
+		log.Errorf("get node info failed: %s", err)
+	}
 	return
 }
 
-// SetPublicKey verify nonce and set Public Key
-func SetPublicKey(id proto.NodeID, nonce mine.Uint256, publicKey *ec.PublicKey) (err error) {
-	keyHash := mine.HashBlock(publicKey.SerializeCompressed(), nonce)
-	idHash, err := hash.NewHashFromStr(string(id))
+// GetAllNodeID
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported function GetAllNodeID should be of the form "GetAllNodeID ..."